x/cronos/keeper/permissions.go

```diff
@@ -0,0 +1,36 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/crypto-org-chain/cronos/v2/x/cronos/types"
+)
+
+// A permission is represented by a bit within uint64 (64bits)
+// An address permission is an integer value between 0 and (2^64-1)
+// This design allows a quick and simple permission check for addresses
+// The next permission should be added before All
+const (
+	CanChangeTokenMapping uint64                                  = 1 << iota // 1
+	CanTurnBridge                                                             // 2
+	All                   = CanChangeTokenMapping | CanTurnBridge             // 3
+)
+
+func (k Keeper) SetPermissions(ctx sdk.Context, address sdk.AccAddress, permissions uint64) {
+	store := ctx.KVStore(k.storeKey)
+	permissionsBytes := sdk.Uint64ToBigEndian(permissions)
+	store.Set(types.AdminToPermissionsKey(address), permissionsBytes)
+}
+
+func (k Keeper) GetPermissions(ctx sdk.Context, address sdk.AccAddress) uint64 {
+	store := ctx.KVStore(k.storeKey)
+	permissionsBytes := store.Get(types.AdminToPermissionsKey(address))
```

Comment:
not sure if we should use `store.Has` with addr and type for less gas

Reply:
mmm not sure if it is worth it considering it simplify the design 

also by storing all the permissions within 64bits, we need only one get to fetch all the permissions rather than iterating over all the types for permissions query